### ISSUE TITLE
[Bug] Incorrect implementation "didFromUrl"

### ISSUE THREAD

**Issue by iliecirciumaru** [interop-alliance/did-web-resolver#20]
If code starts with `verificationMethod` of format:
```
did:web:ex.github.io:qa:alice#mdrZ9
```
Then function `urlFromDid` will return `http://ex.github.io/qa/alice/did.json` what is correct.
However `didFromUrl` having input as `http://ex.github.io/qa/alice/did.json` will return `did:web:ex.github.io:qa:alice:did.json`, what is not correct.
Expected `did` must be `did:web:ex.github.io:qa:alice`.
This happens because these lines have not entirely correct logic:
```
  if (pathname && pathname !== '/' && pathname !== '/.well-known/did.json') {
    pathComponent = pathname.split('/').map(encodeURIComponent).join(':');
  }  
```

Before these lines I would do a transformation (which would fix the problem):
```
if (pathname.endsWith('/did.json')) {
    pathname = pathname.substring(0, pathname.length - '/did.json'.length);
}
```

**Comment by dmitrizagidulin** [comment]
Thanks @iliecirciumaru! You're absolutely right.
Do you mind adding a PR for this?

**Comment by iliecirciumaru** [comment]
> Thanks @iliecirciumaru! You're absolutely right. Do you mind adding a PR for this?

Yes, with a pleasure. 
I will try to do it this week.

**Comment by iliecirciumaru** [comment]
@dmitrizagidulin please take a look when you have time: https://github.com/interop-alliance/did-web-resolver/pull/21

**Comment by dmitrizagidulin** [comment]
@iliecirciumaru - thanks again for the fix! Will release a new version with it, shortly.

**Comment by dmitrizagidulin** [comment]
@iliecirciumaru - published as `v3.0.1`.

**Comment by iliecirciumaru** [comment]
Thanks!
